test/requests/atom_feed_test.rb

```diff
@@ -15,7 +15,9 @@ def app
   end
 
   test "atom feed returns when no gems are defined" do
-    get "/atom.xml"
+    silence_stream($stdout) do
+      get "/atom.xml"
+    end
```

Comment:
I do not see anything improves with this change. Being unchanged looks enough to me.
```suggestion
    get "/atom.xml"
```

Reply:
The difference is that when I run the tests, I don't really want to see output from the program, just test results. However, I do want to see errors.

Reply:
How different are they before and after this change then?

Reply:
Before we continue the discussion of this particular change here, I would like to clarify that we're on the same page.

Do you agree with the goal that a successful unit test run should ideally produce nothing but (green) dots when run with compact test report, i.e. not output from the tested program? And that suppressing error output might have the undesired side effect of not detecting that something is amiss?

This commit achieves this goal. If you don't agree, I can do nothing but revert the change.

Reply:
And here is the difference:

Before
````
$ rake test:units 

# Running tests with run options --seed 52198:

..............WARNING:  no files specified
WARNING:  See https://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: example
  Version: 1.0.0
  File: example-1.0.0.gem
....................................WARNING:  no files specified
WARNING:  See https://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: example
  Version: 1.0.0
  File: example-1.0.0-x86_64-linux.gem
...........................

Finished tests in 1.008565s, 76.3461 tests/s, 118.9809 assertions/s.


77 tests, 120 assertions, 0 failures, 0 errors, 0 skips
````

After:
````
$ rake test:units 

# Running tests with run options --seed 31791:

.............................................................................

Finished tests in 0.183664s, 419.2438 tests/s, 653.3670 assertions/s.


77 tests, 120 assertions, 0 failures, 0 errors, 0 skips
````

Reply:
But wait a sec, I think there's something going on I don't quite understand yet, because now I can't reproduce the output. I'll get back to you later. In the meantime, I'm going back to draft.

Reply:
What's going on is that the directories created during test runs are not properly cleaned up. This is a topic for later.

I think I found a better way of achieving the same result. Please check the latest commit on the PR.

Reply:
> And here is the difference:

This was NOT from changes around `get "/atom.xml"`, but from changes in `test/requests/is_api_request_test.rb`. But finally you found a better way later, which looks so nice! 🕶️